core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -168,8 +170,20 @@ protected boolean isClassExcluded(Class<?> clazz) {
             return true;
         }
         for (Class<?> excludedClass : excludedClasses) {
-            if (clazz.isAssignableFrom(excludedClass)) {
-                return true;
+            if(excludedClass != Object.class) {
+                if(clazz.getName().startsWith(packageName)){
```

Comment:
Why do you need this check? The whole `com.opensymphony.xwork2.ognl.` package is already excluded and [package exclusion is tested before class exclusion](https://github.com/apache/struts/pull/546/files#diff-bd4f5dc22219d389a455107a004139c3fa3a94b9493946dc9b0c927f26217853L85-L99)

Reply:
if i don't add this check, the unit test can't pass.

Reply:
Also this change make that all the tests have passed as they are located in `com.opensymphony.xwork2.ognl.`

Reply:
the old check in com.opensymphony.xwork2.ognl.   isn't change , also use  clazz.isAssignableFrom(excludedClass)
the new check use excludedClass.isAssignableFrom(clazz)

Reply:
Yes, basically you adjusted the code just to satisfy tests, it will break users' applications as they do not use `com.opensymphony.xwork2.ognl.` in their code.

Reply:
this change is for safety.
many other class can replace #@org.apache.commons.collections.BeanMap@{} to clean excludedPackageNames and excludedClasses.
we can't allow excludedClasses geting longer and longer.

Reply:
This exact change is not for safety, it's to omit tests and it degrades our _safety net_. Could you provide a use case for you PR? I would like to implement an unit test to cover those changes.

Reply:
I just copy/paste `SecurityMemberAccessTest` into a custom package (eg. `com.test`) to avoid messing with `com.opensymphony.xwork2.ognl.` package and to double check your change - it fails on `testMiddleOfInheritanceExclusion1`.

Reply:
ok , i give up. =)
i am just web security guy , i don't good at coding